src/openrc-run/openrc-run.c

```diff
@@ -423,61 +426,62 @@ svc_exec(const char *arg1, const char *arg2)
 		/* UNREACHABLE */
 	}
 
-	buffer = xmalloc(sizeof(char) * BUFSIZ);
 	fd[0].fd = signal_pipe[0];
+	fd[1].fd = master_tty;
 	fd[0].events = fd[1].events = POLLIN;
 	fd[0].revents = fd[1].revents = 0;
-	if (master_tty >= 0) {
-		fd[1].fd = master_tty;
-		fd[1].events = POLLIN;
-		fd[1].revents = 0;
-	}
 
+	warn_timeout = clock_ms() + (WARN_TIMEOUT * 1000);
```

Comment:
Slight nitpick:

```c
now = clock_ms();
wait_timeout = now + (WAIT_TIMEOUT * 1000);
warn_timeout = now + (WARN_TIMEOUT * 1000);
... = fork();
```

Is more consistent than doing two different `clock_ms()`. Also has the added benefit of both of these being set near each other.